components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3648,10 +3657,10 @@ subroutine compute_rain_fall_velocity(qr_incld, rcldm, rhofacr, nr, nr_incld, mu
 
    call find_lookupTable_indices_3(dumii,dumjj,dum1,rdumii,rdumjj,inv_dum3,mu_r,lamr)
 
-   !nr is not used elsewhere in this function. Here, the updated value from generalized_sed
-   !is getting overwritten with the incld value which wasn't getting updated until this PR.
-   !And why is nr getting updated but not qr? Should delete nr from this function entirely.
-   !nr = nr_incld*rcldm 
+   !get_rain_dsd2 keeps the drop-size distribution within reasonable
+   !bounds by modifying nr_incld. The next line maintains consistency
+   !between nr_incld and nr
+   nr = nr_incld*rcldm 
```

Comment:
@PeterCaldwell 
I would get this update for `nr` outside of this routine (i.e., erase `nr` inside this routine), and place it at the scope similar to the update of `nc` (see above).
Otherwise, the Fortran side looks OK and I'm finalizing the review.

Reply:
Good idea to update nr outside compute_rain_fall_velocity in order to simplify interfaces and improve clarity. I've done that now.